wasm/jit/jit_amd64.go

```diff
@@ -452,8 +483,12 @@ func (c *amd64Compiler) compileBr(o *wazeroir.OperationBr) error {
 	if err := c.maybeMoveTopConditionalToFreeGeneralPurposeRegister(); err != nil {
 		return err
 	}
+	return c.branchInto(o.Target)
+}
 
-	if o.Target.IsReturnTarget() {
+// branchInto adds instruction necessary to jump into the given branch target.
```

Comment:
I think it is fine to not prefix with compile because the param isn't an operation